compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -238,6 +238,7 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
             && has_changed
             && !self.in_projection_eq_hack
             && !self.search_graph.in_cycle()
+            && false
```

Comment:
Both of these trait solvers fail here. That's because the inference guidance from a `Certainty::AMBIGUOUS` response can make the goal pass the next time it's evaluated.

I think that that's fine, though? We probably just need to use `evaluate_goals` everywhere in the solver, so we process them until they are unchanged...